### ISSUE TITLE
Connections to other network are not taken into account in integrity check

### ISSUE THREAD

**Issue by A67-A67** [vmarc/knooppuntnet#32]
For the integrity check only the non-connection routes are taken into account. This results in an false-positive at nodes near the edge of a network.

Example: [https://knooppuntnet.nl/nl/node/42585285](https://knooppuntnet.nl/nl/node/42585285)
Node 6 is in the network of Geldrop-Mierlo. It should have three routes according to the tag expected_rwn_route_relations. Indeed this node has three routes (2-6, 6-27 and 6-50).

However, route 6-50 is a connecting route to node 50 in the network of Heeze-Leende. Therefore it seems that it is not taken into account for the integrity check. This brings the number of routes to two (2-6 and 6-27) and the integrity check fails, even when it is tagged correctly.

In short, the integrity check should be done in the network of which the node is a member (Geldrop-Mierlo in this case, not Heeze-Leende), but it should take all routes of this network into account, also the routes with role connection.

**Comment by vmarc** [comment]
@A67-A67 Many thanks for the clear explanation of the problem.

The logic for counting routes was based on the description in the wiki page: [Cycle Node Network Tagging](https://wiki.openstreetmap.org/wiki/Cycle_Node_Network_Tagging).

> _Count only regular routes, route relations with state=connection or state=alternate are not counted towards the number of expected route relations._

The mistake that was made, was that beside not counting routes with _state=connection_, also routes with _role connection_ in the network relation were not counted.

The code in v2.0.12 for [NodeIntegrityAnalyzer](https://github.com/vmarc/knooppuntnet/blob/f3f9564c0859b9dc44cc17154fc0b3ed2e449ebc/production/core/src/main/scala/kpn/core/engine/analysis/NodeIntegrityAnalyzer.scala) was [changed](https://github.com/vmarc/knooppuntnet/commit/f3f9564c0859b9dc44cc17154fc0b3ed2e449ebc#diff-6d58cba81da3e310d7683b23336a6c64)  to no longer exclude routes with role connection.

This change drastically reduces the number of integrity check failures reported!
